pandas/core/arrays/base.py

```diff
@@ -757,13 +757,20 @@ def astype(self, dtype: AstypeArg, copy: bool = True) -> ArrayLike:
         >>> arr2.dtype
         dtype('float64')
         """
+        from pandas.api.types import CategoricalDtype
+
         dtype = pandas_dtype(dtype)
         if dtype == self.dtype:
             if not copy:
                 return self
             else:
                 return self.copy()
 
+        if isinstance(dtype, CategoricalDtype):
```

Comment:
Pretty weird to do this in the base class.  Can you track down where the actual problem is? Could be hiding other bugs eg in categorical.fromsequence